src/luarocks/core/cfg.lua

```diff
@@ -106,15 +106,19 @@ do
    end
 
    -- Merge values from config files read into the `cfg` table
-   local function merge_overrides(cfg, overrides)
+   local function merge_overrides(cfg, overrides, hardcoded)
```

Comment:
I believe such low-level function should not know something about `hardcoded` specific. Higher level functions should decide what to overwrite/merge.

Reply:
Now "overrides" merges into "hardcoded", then "hardcoded" merges into "cfg". 
` util.deep_merge(hardcoded,result)
         merge_overrides(cfg.variables, hardcoded)
         return filepath`